tern/formats/default/generator.py

```diff
@@ -42,18 +41,27 @@ def print_full_report(image):
                 if package.pkg_license not in layer_license_list and \
                         package.pkg_license:
                     layer_license_list.append(package.pkg_license)
-                    if package.pkg_license not in full_license_list:
-                        full_license_list.append(package.pkg_license)
             # Collect packages + licenses in the layer
             notes = notes + formats.layer_packages_list.format(
                 list=", ".join(layer_pkg_list) if layer_pkg_list else 'None')
             notes = notes + formats.layer_licenses_list.format(list=", ".join(
                 layer_license_list) if layer_license_list else 'None')
             notes = notes + formats.package_demarkation
+    return notes
+
+
+def print_licenses_only(image_obj_list):
+    '''Print a complete list of licenses for all images'''
+    full_license_list = []
+    for image in image_obj_list:
+        for layer in image.layers:
+            for package in layer.packages:
+                if package.pkg_license not in full_license_list:
+                    full_license_list.append(package.pkg_license)
```

Comment:
@nishakm When I test these changes with a debian image that does not provide license information I don't see the `None` description under the summary:

```
        Licenses found in Layer:  None
------------------------------------------------

###########################################
# Summary of licenses found in Container: #
###########################################

```

I wonder if we might want to keep the `None` there at the bottom summary as a blank line can be somewhat confusing. It kind of looks like the license list just forgot to get printed when it is blank. What do you think?

Reply:
Ah good catch! I will fix this.